src/background/browser-adapter.ts

```diff
@@ -94,7 +94,11 @@ export class ChromeAdapter extends ClientChromeAdapter implements BrowserAdapter
     }
 
     public getTab(tabId: number, callback: (tab: chrome.tabs.Tab) => void): void {
```

Comment:
I don't think this is an appropriate error contract; if I'm trying to call `getTab` from somewhere else in the code, and I call it with an invalid tab, I want the callsite to be responsible for deciding whether it's acceptable to allow that and move on or propogate an error with more context.

I think the ideal way to handle that would be to make this an async function and emit an exception that the caller can choose to handle or not, but if we don't want to do that for just this one particular function, I'd do it with the javascript-typical resolve/reject callback pattern.

Reply:
Great point, I'm also thinking doing that.